### ISSUE TITLE
InvalidTransition#message refers to machine's current state, not the state that caused the error

### ISSUE THREAD

**Issue by D-side** [aasm/aasm#312]
About this line: https://github.com/aasm/aasm/blob/master/lib/aasm/errors.rb#L13

We just had a day-long headache thinking over a weird error message we were getting from a pretty simple state machine (mostly truncated of unnecessary details):

```ruby
state :scheduled, :initial => true
state :running
state :error
state :success

event :retry do
  transitions to: :running, from: [:error]
end

event :fail do
  transitions to: :error, from: [:running, :scheduled, :error, :success]
end
```

We were getting the following in our error collection system:

    Event 'retry' cannot transition from 'error'

...and we were like **"wat?"** The state machine explicitly specified otherwise! Well... we actually had a bug in our app and fixed it. Here's what happened:

* We actually got an instance with a `scheduled` state. We attempted to `retry` it and got an `InvalidTransition`.
* Falling down into a `rescue` block, we `fail`ed the state machine (changing its state to `error`).
* We logged the exception into our collection system (`exception.message` and `exception.backtrace`)... except its message "specified the impossible" (given above) as it was referencing the machine's current state, `error`! Whereas when it happened the state was actually `scheduled`.

**Ignoring the shady programming practices one could detect above, is this the expected behavior? Shouldn't `InvalidTransition#message` indicate what the state was when it was raised, instead of using machine's current state?**

**Comment by alto** [comment]
@D-side I agree. Which version of _AASM_ are you using?

**Comment by D-side** [comment]
Hi, @alto, thanks for responding and sorry for the delay.

As seen in the link above, the problematic code is in the `master` branch, so...

FWIW, we're using **aasm 4.3.0**. We've worked around this (by logging exception right away, which we should have done initially), we just had a bit of a hard time identifying the issue. Following the "principle of least surprise" you might want to fix this in future releases. Thanks :)

**Comment by alto** [comment]
I just released version `4.10.0` including the changes from #341.
